src/components/WebexAdaptiveCard/WebexAdaptiveCard.jsx

```diff
@@ -5,24 +5,47 @@ import webexComponentClasses from '../helpers';
 import {useAdaptiveCard} from '../hooks';
 import AdaptiveCard from '../adaptive-cards/AdaptiveCard/AdaptiveCard';
 
+/**
+ * Action to perform when submitting a card
+ *
+ * @name onSubmit
+ * @function
+ * @param {Promise<object|string} submitPromise  Adapter submit activity or an error
+ * @param {object} inputs Data to submit
+ */
+
 /**
  * WebexAdaptiveCard component displays a webex adaptive card.
  *
  * @param {object} props  Data passed to the component
  * @param {string} props.activityID  ID of the activity corresponding to this card
  * @param {string} [props.className]  Custom CSS class to apply
  * @param {object} [props.style]  Custom style to apply
+ * @param {Function} [props.onSubmit]  Action to perform when submitting a card
  * @returns {object} JSX of the component
  */
-export default function WebexAdaptiveCard({activityID, className, style}) {
+export default function WebexAdaptiveCard({
+  activityID,
+  className,
+  style,
+  onSubmit,
+}) {
   const [card, submit] = useAdaptiveCard(activityID);
   const [cssClasses] = webexComponentClasses('adaptive-card', className);
 
+  const handleSubmit = (inputs) => {
+    const submitPromise = new Promise((resolve, reject) => {
+      submit(inputs).subscribe(resolve, reject);
```

Comment:
I just realized there's a simpler way to achieve this, using the toPromise() operator